lib/util/git/index.ts

```diff
@@ -395,6 +398,34 @@ export async function isBranchStale(branchName: string): Promise<boolean> {
   return !branches.all.map(localName).includes(branchName);
 }
 
+export async function isBranchModified(branchName: string): Promise<boolean> {
+  // First check cache
+  if (config.branchIsModified[branchName] !== undefined) {
+    return config.branchIsModified[branchName];
+  }
+  if (!(await branchExists(branchName))) {
+    throw Error(
+      'Cannot check modification for branch that does not exist: ' + branchName
+    );
+  }
+  // Retrieve the author of the most recent commit
+  const lastAuthor = (
```

Comment:
Can we also count the commits? Assume a user reordered his commit before the renovate commit.

Reply:
I'm not sure there's an easy way for us to do that at the git level. We might need to know what the target sha was in order to know when to stop counting commits.

Reply:
Possible

https://stackoverflow.com/a/11657647

```sh
git checkout master
git checkout -b test
<We do 3 commits>
git rev-list --count HEAD ^master
``|

Reply:
`git rev-list --count develop..renovate/branch` should do it. Not yet tried

Reply:
It would be easy if it's up to date with master. I'm worried about the case where it branched off a few commits ago

Reply:
Maybe works:
```
❯ git rev-list --count fix/5951-retry-checkbox ^master
1
```

(the above branch was already merged and is out of date)